### PR TITLE
Add admin page for keys.

### PR DESCRIPTION
**Pull request by Acidity**
Signed-off-by: Tyler O'Meara Tyler@TylerOMeara.com

### CODE REVIEW

File: brave/core/key/template/list.html [hunk 1]
```diff
@@ -148,18 +148,25 @@
...
-        <td class="key-code"><span class="ellipsis">${record.code | h}</span></td>
         % if record.violation == "Character":
         <td class="security-violation"><font color="red">Character Security Violation.</font></td>
         % else:
         <td class="security-violation"><font color="green">The BIA is always watching...</font></td>
         % endif
+        % if not admin:
+        <td class="key-code"><span class="ellipsis">${record.code | h}</span></td>
+        % else:
+        <td class="key-code" title="Thanks Draleth."><span class="ellipsis">Redacted.</span></td>
```
Comment (jboning) [inline brave/core/key/template/list.html]: wat

File: brave/core/key/controller.py [hunk 2]
```diff
@@ -78,11 +78,17 @@ def get(self, admin=False):
         
         if admin and not is_administrator:
             raise HTTPNotFound()
+            
+        credentials = user.credentials
+        if admin:
+            #Don't send the verification code for the API keys.
+            #See: The Great Brave API Key Leak of 2014
```
Comment (jboning) [inline brave/core/key/controller.py]: I think why is obvious enough, we don't need to go pointing fingers at specific people :P

File: brave/core/key/template/list.html [hunk 3]
```diff
@@ -203,6 +210,9 @@
                         <th class="span3 sortable key-code"><span class="line"></span>${_("Verification Code")}</th>
                         <th class="span2 sortable characters hidden-tablet hidden-phone"><span class="line"></span>${_("Characters")}</th>
                         <th class="span3 sortable key-date hidden-phone"><span class="line"></span>${_("Added")}</th>
+                        % if admin:
+                        <th class="span3 sortable username"><span class="line"></span>${_("User")}</th>
```
Comment (jboning) [inline brave/core/key/template/list.html]: the widths don't add up right in admin mode

Comment (Acidity) [inline brave/core/key/template/list.html]: I'm not exactly sure what you mean by this (I don't use HTML often).

Comment (jboning) [inline brave/core/key/template/list.html]: The bootstrap "span*" things are supposed to sum to something particular, right? With ?admin=1, the table is really wide (overflows the page for me).

Comment (Acidity) [inline brave/core/key/template/list.html]: Well I thought it was supposed to be 12, but last time I modified a page to be 12 you questioned that :P

I'll go back and make both this and the other one have a width of 12, and see if that fixes it. I have no problems when viewing it in chromium as is though.

Comment (jboning) [inline brave/core/key/template/list.html]: Oh. My question last time was "what's this about", and you had a good answer :P

**Pull request merged**